lib/SabreAMF/Message.php

```diff
@@ -148,14 +149,14 @@ public function deserialize(SabreAMF_InputStream $stream) {
                 if (is_object($body['data']) && $body['data'] instanceof SabreAMF_AMF3_Wrapper) {
                      $body['data'] = $body['data']->getData();
                      $this->encoding = SabreAMF_Const::AMF3;
-                } else if (is_array($body['data']) && isset($body['data'][0]) && is_object($body['data'][0]) && $body['data'][0] instanceof SabreAMF_AMF3_Wrapper) {
-
-                     if ( defined('SABREAMF_AMF3_PRESERVE_ARGUMENTS') ) {
```

Comment:
This looks like you're removing a feature?

Reply:
I'm not really deleting a function, due to SabreAMF bug, if it detects an object on the first element of the array it will just return that object and ignore rest of elements, ill look into it soon to see if im not wrong